vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -92,48 +93,95 @@ void computeSpatialSourceCellsForPencil(const dccrg::Dccrg<SpatialCell,dccrg::Ca
    // L = length of the pencil iPencil
    int L = pencils.lengthOfPencils[iPencil];
    vector<CellID> ids = pencils.getIds(iPencil);
-   
-   int neighborhood = getNeighborhood(dimension,2);
+
+   // These neighborhoods now include the AMR addition beyond the regular vlasov stencil
+   int neighborhood = getNeighborhood(dimension,VLASOV_STENCIL_WIDTH);
 
    // Get pointers for each cell id of the pencil
    for (int i = 0; i < L; ++i) {
       sourceCells[i + VLASOV_STENCIL_WIDTH] = mpiGrid[ids[i]];
    }
-   
+
    // Insert pointers for neighbors of ids.front() and ids.back()
    const auto* frontNbrPairs = mpiGrid.get_neighbors_of(ids.front(), neighborhood);
    const auto* backNbrPairs  = mpiGrid.get_neighbors_of(ids.back(),  neighborhood);
 
    int maxRefLvl = mpiGrid.get_maximum_refinement_level();
-      
+
    // Create list of unique distances in the negative direction from the first cell in pencil
    std::set< int > distances;
    for (const auto nbrPair : *frontNbrPairs) {
       if(nbrPair.second[dimension] < 0) {
-         distances.insert(nbrPair.second[dimension]);
+         // gather positive distance values
+         distances.insert(-nbrPair.second[dimension]);
       }
    }
 
-
    int iSrc = VLASOV_STENCIL_WIDTH - 1;
+
    // Iterate through distances for VLASOV_STENCIL_WIDTH elements starting from the smallest distance.
-   // Distances are negative here so largest distance has smallest value
-   auto irend = distances.rbegin();
-   std::advance(irend, std::min((int)distances.size(), VLASOV_STENCIL_WIDTH));
-   for (auto it = distances.rbegin(); it != irend; ++it) {
+   for (auto it = distances.begin(); it != distances.end(); ++it) {
+      if (iSrc < 0) break; // found enough elements
+
       // Collect all neighbors at distance *it to a vector
       std::vector< CellID > neighbors;
       for (const auto nbrPair : *frontNbrPairs) {
-         int distanceInRefinedCells = nbrPair.second[dimension];
+         int distanceInRefinedCells = -nbrPair.second[dimension];
          if(distanceInRefinedCells == *it) neighbors.push_back(nbrPair.first);
       }
+      // Get rid of duplicate neighbor cells at single distance
+      neighbors.erase(unique(neighbors.begin(), neighbors.end()), neighbors.end());
 
       int refLvl = mpiGrid.get_refinement_level(ids.front());
-      
       if (neighbors.size() == 1) {
+         if (sourceCells[iSrc+1] == mpiGrid[neighbors.at(0)]) continue; // already found this cell for different distance         
          sourceCells[iSrc--] = mpiGrid[neighbors.at(0)];
       } else if ( pencils.path[iPencil][refLvl] < neighbors.size() ) {
+         if (sourceCells[iSrc+1] == mpiGrid[neighbors.at(pencils.path[iPencil][refLvl])]) continue; // already found this cell for different distance (should not happen)
          sourceCells[iSrc--] = mpiGrid[neighbors.at(pencils.path[iPencil][refLvl])];
+         // Code to verify that multiple neighbors are in correct ordering (z-y-x)
```

Comment:
Add this block behind a `if(debug)` or so? We have a few of those in this file anyway.